lib/rubygems/errors.rb

```diff
@@ -170,12 +170,21 @@ def initialize(source, error)
     # An English description of the error.
 
     def wordy
-      "Unable to download data from #{@source.uri} - #{@error.message}"
+      redact_password("Unable to download data from #{@source.uri} - #{@error.message}", @source.uri)
```

Comment:
@segiddins I pushed up a new patch redacting the password here and in FetchError.